lib/puppet/type/ec2_vpc_vpn.rb

```diff
@@ -0,0 +1,62 @@
+Puppet::Type.newtype(:ec2_vpc_vpn) do
+  @doc = 'Type representing an AWS Virtual Private Networks.'
+
+  ensurable
+
+  newparam(:name, namevar: true) do
+    desc 'The name of the VPN.'
+    validate do |value|
+      fail 'VPNs must have a name' if value == ''
+    end
+  end
+
+  newproperty(:vpn_gateway) do
+    desc 'The VPN gateway to attach to the VPN.'
+  end
+
+  newproperty(:customer_gateway) do
+    desc 'The customer gateway to attach to the VPN.'
+  end
+
+  newproperty(:type) do
+    desc 'The type of VPN gateway, defaults to ipsec.1.'
+    defaultto 'ipsec.1'
+    validate do |value|
+      unless value =~ /^ipsec\.1$/
```

Comment:
Why is this a writeable property if there is only one valid value? This should either not be a property or be a read-only property to reflect how AWS reports it.